.github/workflows/lint.yml

```diff
@@ -0,0 +1,23 @@
+name: Lint on pull requests (or manually)
+on:
+  pull_request:
+  workflow_dispatch:
+jobs:
+  docs:
+    runs-on: ubuntu-latest
+    name: PR - Markdown lint
+    steps:
+      - uses: actions/checkout@v2
+      - name: Run markdownlint
+        uses: actionshub/markdownlint@2.0.2
+  actions:
+
+    runs-on: ubuntu-latest
+    name: PR - Action lint
+    steps:
+      - uses: actions/checkout@v2
+      - run: |
+          echo "::add-matcher::.github/actionlint-matcher.json"
+          bash <(curl https://raw.githubusercontent.com/rhysd/actionlint/main/scripts/download-actionlint.bash)
+          ./actionlint -color -shellcheck=
+        shell: bash
```

Comment:
I'd probably whack in Hadolint as well now just because it is easy and will do nothing if no Dockerfiles
```
  hadolint-pr:
    runs-on: ubuntu-latest
    name: PR - Hadolint
    steps:
      - uses: actions/checkout@v2
        # Ignores do not work: https://github.com/reviewdog/action-hadolint/issues/35 is resolved
      - uses: reviewdog/action-hadolint@v1
```

Reply:
Oh, and MarkdownLint